python/lib/sift_py/data_import/ch10.py

```diff
@@ -0,0 +1,111 @@
+import json
+from typing import Any, Dict, Optional
+
+import requests
+
+from sift_py.data_import.config import CsvConfig
+from sift_py.data_import.csv import CsvUploadService
+from sift_py.data_import.status import DataImportService
+from sift_py.data_import.time_format import TimeFormatType
+
+
+class BaseCh10File:
+    """
+    Base class for uploading IRIG Chapter 10/Chapter 11 files.
+
+    Implement a concrete version of this class that parses a ch10 stream and returns
```

Comment:
I think it would be help to add type hints to the `__iter__` and `__next__` functions to make it very clear how this base class should be implemented.

Reply:
Also i think an inline example showing how to implement this class would help remarkably. Checkout `lib/sift_py/ingestion/service.py` -> `buffered_ingestion`.